### PR TITLE
feat: add readme for subgraphs and federated graphs

### PR DESCRIPTION
**Pull request by JivusAyrus**
- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

image
image

image

image

image

### CODE REVIEW

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 1]
```diff
@@ -191,6 +194,7 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
           createdBy: authContext.userId,
           labelMatchers: req.labelMatchers,
```
Comment (StarpTech) [inline controlplane/src/core/bufservices/PlatformService.ts:195]: Can we generate a default README when creating a graph for the first time?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: cli/src/commands/federated-graph/commands/create.ts [hunk 2]
```diff
@@ -20,12 +23,19 @@ export default (opts: BaseCommandOptions) => {
     '--label-matcher [labels...]',
     'The label matcher is used to select the subgraphs to federate. The labels are passed in the format <key>=<value> <key>=<value>. They are separated by spaces and grouped using comma. Example: --label-matcher team=A,team=B env=prod',
   );
+  createFederatedGraph.option('--readme <path-to-readme>', 'The markdown file which describes the federated graph.');
   createFederatedGraph.action(async (name, options) => {
+    let readmeFile;
+    if (options.readme) {
+      readmeFile = resolve(process.cwd(), options.readme);
```
Comment (StarpTech) [inline cli/src/commands/federated-graph/commands/create.ts:30]: Please check if the file exists, if not throw an error.

File: cli/src/commands/federated-graph/commands/update.ts [hunk 3]
```diff
@@ -17,12 +20,19 @@ export default (opts: BaseCommandOptions) => {
...
+    if (options.readme) {
+      readmeFile = resolve(process.cwd(), options.readme);
+    }
+
     const resp = await opts.client.platform.updateFederatedGraph(
       {
         name,
         routingUrl: options.routingUrl,
         labelMatchers: options.labelMatcher,
+        readme: readmeFile && existsSync(readmeFile) ? await readFile(readmeFile, 'utf8') : undefined,
```
Comment (StarpTech) [inline cli/src/commands/federated-graph/commands/update.ts:35]: Here too

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: cli/src/commands/federated-graph/commands/create.ts (hunk 2 above)
Comment (JivusAyrus) [reply cli/src/commands/federated-graph/commands/create.ts:30]: Done

File: cli/src/commands/federated-graph/commands/update.ts (hunk 3 above)
Comment (JivusAyrus) [reply cli/src/commands/federated-graph/commands/update.ts:35]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**